applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_hydraulic_solver.py

```diff
@@ -0,0 +1,692 @@
+                "element_type" : "levelset_convection_supg",
+                "element_settings" : {
+                    "dynamic_tau" : 1.0,
+                    "tau_nodal":true
+                }
+            },
+            "levelset_convection_settings": {
+                "max_CFL" : 1.0,
+                "max_substeps" : 0,
+                "eulerian_error_compensation" : false,
+                "element_type" : "levelset_convection_supg",
+                "element_settings" : {
+                    "dynamic_tau" : 1.0,
+                    "tau_nodal":true
+                }
+            },
+            "distance_reinitialization": "variational",
+            "parallel_redistance_max_layers" : 25,
+            "distance_smoothing": false,
+            "distance_smoothing_coefficient": 1.0,
+            "distance_modification_settings": {
+                "model_part_name": "",
+                "distance_threshold": 1e-5,
+                "continuous_distance": true,
+                "check_at_each_time_step": true,
+                "avoid_almost_empty_elements": false,
+                "deactivate_full_negative_elements": false
+            }
+        }""")
+
+        default_settings.AddMissingParameters(super(NavierStokesTwoFluidsHydraulicSolver, cls).GetDefaultParameters())
+        return default_settings
+
+    def __init__(self, model, custom_settings):
+        # TODO: DO SOMETHING IN HERE TO REMOVE THE "time_order" FROM THE DEFAULT SETTINGS BUT KEEPING THE BACKWARDS COMPATIBILITY
+
+        if custom_settings.Has("levelset_convection_settings"):
+            if custom_settings["levelset_convection_settings"].Has("levelset_splitting"):
+                custom_settings["levelset_convection_settings"].RemoveValue("levelset_splitting")
+                KratosMultiphysics.Logger.PrintWarning("NavierStokesTwoFluidsHydraulicSolver", "\'levelset_splitting\' has been temporarily deactivated. Using the standard levelset convection with no splitting.")
+
+        #TODO: Remove this after the retrocompatibility period
+        if custom_settings.Has("bfecc_convection"):
+            KratosMultiphysics.Logger.PrintWarning("NavierStokesTwoFluidsHydraulicSolver", "the semi-Lagrangian \'bfecc_convection\' is no longer supported. Using the standard Eulerian levelset convection.")
+            custom_settings.RemoveValue("bfecc_convection")
+            if custom_settings.Has("bfecc_number_substeps"):
+                custom_settings.RemoveValue("bfecc_number_substeps")
+
+        super(NavierStokesTwoFluidsHydraulicSolver,self).__init__(model,custom_settings)
+
+
+        self.element_name = "TwoFluidNavierStokesAlphaMethod"
+        self.rho_inf = 0.0
+        self.main_model_part.ProcessInfo.SetValue(KratosCFD.SPECTRAL_RADIUS_LIMIT, self.rho_inf)
+        self.condition_name = "TwoFluidNavierStokesWallCondition"
+        self.element_integrates_in_time = True
+        self.element_has_nodal_properties = True
+
+        self.min_buffer_size = 3
```

Comment:
Isn't 2 the buffer required for the alpha scheme?

Reply:
Oups, Yes, now it is modified.